examples/examples/z_querier.rs

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use std::time::Duration;
+
+use clap::Parser;
+use zenoh::{key_expr::KeyExpr, query::QueryTarget, Config};
+use zenoh_examples::CommonArgs;
+
+#[tokio::main]
+async fn main() {
+    // initiate logging
+    zenoh::init_log_from_env_or("error");
+
+    let (config, keyexpr, payload, target, timeout) = parse_args();
+
+    println!("Opening session...");
+    let session = zenoh::open(config).await.unwrap();
+
+    println!("Declaring Querier on '{keyexpr}'...");
+    let querier = session
+        .declare_querier(keyexpr)
+        .target(target)
+        .timeout(timeout)
+        .await
+        .unwrap();
+
+    #[cfg(feature = "unstable")]
+    querier
+        .matching_listener()
+        .callback(|matching_status| {
```

Comment:
I'm not sure, to me it seems it would make code more difficult to read (especially given that I would need to introduce conditionals based on enabled features both inside and outside of tokio::select. @Mallets do you have any opinion on it ?

Reply:
It's indeed not trivial. It's good enough for now. Let's forget that.